### PR TITLE
Add support for Lenovo Smart Tab M10

### PR DESCRIPTION
**Pull request by superna9999**
Base DT is already merged for v6.11:
https://lore.kernel.org/all/20240523-topic-sdm450-upstream-tbx605f-v1-0-e52b89133226@linaro.org/

Panel has been submitted and is under review:
https://lore.kernel.org/all/20240709-topic-sdm450-upstream-tbx605f-panel-v1-0-af473397835d@linaro.org/

Panel driver depends on the "mipi_dsi wrapped functions" that were merged in drm-misc-next for v6.11, and were backported to allow the panel driver to properly build.

The battery & panel nodes are not upstreamed yet due to to the lack of smb charger and lab/ibb upstream support.

An experimental PostmarketOS changeset is available at https://gitlab.com/superna9999/pmaports/-/commits/lenovo-tbx605f

### CODE REVIEW

File: drivers/gpu/drm/panel/panel-boe-tv101wum-ll2.c [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+	mipi_dsi_set_drvdata(dsi, ctx);
+
+	dsi->lanes = 4;
+	dsi->format = MIPI_DSI_FMT_RGB888;
+	dsi->mode_flags = MIPI_DSI_MODE_VIDEO | MIPI_DSI_MODE_VIDEO_BURST |
+			  MIPI_DSI_MODE_VIDEO_HSE;
+
+	drm_panel_init(&ctx->panel, dev, &boe_tv101wum_ll2_panel_funcs,
+		       DRM_MODE_CONNECTOR_DSI);
+	ctx->panel.prepare_prev_first = true;
```
Comment (barni2000) [inline drivers/gpu/drm/panel/panel-boe-tv101wum-ll2.c:194]: This should be removed, the repo has some WIP dsi patches.

**Comment by barni2000** [comment]
> The battery & panel nodes are not upstreamed yet due to to the lack of smb charger and lab/ibb upstream support.

You can send panel nodes upstream without lab/ibb (in most cases bootloader is enabling them)

File: drivers/gpu/drm/panel/panel-boe-tv101wum-ll2.c (hunk 1 above)
Comment (barni2000) [reply drivers/gpu/drm/panel/panel-boe-tv101wum-ll2.c:194]: https://github.com/msm8953-mainline/linux/commit/ac8fced2c6734b2d897dbc2e7e09f24156701f64

Comment (superna9999) [reply drivers/gpu/drm/panel/panel-boe-tv101wum-ll2.c:194]: Good, but my goal is to have it all upstream, and it works fine with `ctx->panel.prepare_prev_first = true;` on mainline and this tree, so why should I alter the patch ?

Comment (barni2000) [reply drivers/gpu/drm/panel/panel-boe-tv101wum-ll2.c:194]: What happens when you suspend and wake up your panel dont you have error messages in dmesg?

Comment (superna9999) [reply drivers/gpu/drm/panel/panel-boe-tv101wum-ll2.c:194]: Right, it does error in susend/resume and indeed https://github.com/msm8953-mainline/linux/commit/ac8fced2c6734b2d897dbc2e7e09f24156701f64 is a welcome fix in this downstream fork, I'll add a patch on top removing this line to fix panel suspend/resume

**Review by barni2000 — APPROVED** [review]

**Comment by barni2000** [comment]
I have merged it on the 6.10/main branch.

**Pull request closed without merging**